datafusion/expr/src/type_coercion/functions.rs

```diff
@@ -293,6 +314,32 @@ fn maybe_data_types(
     Some(new_type)
 }
 
+/// Check if the current argument types can be coerced to match the given `valid_types`
+/// unlike `maybe_data_types`, this function does not coerce the types.
```

Comment:
this comment says "function does not coerce the types" but then it permits types in `can_cast_types`. I may be misunderstanding what coerce means in this case

Reply:
I think the role of `can_cast_types` is more like validator to ensure the valid_types given are castable for arrow-cast. Maybe we don't even need this 🤔